pages/api/register.ts

```diff
@@ -1,26 +1,34 @@
-import { NextApiRequest, NextApiResponse } from "next"
-import { client } from "../../lib/graphql"
+import { NextApiHandler } from "next";
+import fetch from "node-fetch";
 
-const handler = async (request: NextApiRequest, res: NextApiResponse): Promise<undefined> => {
-  const saleor_domain = request.headers['x-saleor-domain']
+const handler: NextApiHandler = async (request, response) => {
+  const saleor_domain = request.headers["saleor-domain"];
```

Comment:
this was also changed, because `x-` prefix is deprecated and since Saleor@4 there will be just `Saleor-Domain` (see: https://docs.saleor.io/docs/3.x/developer/extending/apps/asynchronous-webhooks#headers)